src/components/Registration.jsx

```diff
@@ -0,0 +1,56 @@
+import React, { useState } from 'react';
+import '../components/Registration.css';
+
+const Registration = () => {
+
+    const [user, setUser] = useState ({
+        username: "",
+        email: "",  
+        password: "",
+        passwordConfirm: "",
+     })
+
+     const handleEmail = (typedEmail) => {
+        if (typedEmail.includes("@")) {
+        setUser({...user, email: typedEmail});
+        }
+     }
+
+     const handleUserName = () => {
+        if (user.email !== null) {
+        const email = user.email;
+        setUser({...user, username: email.slice(0, email.indexOf("@"))})}
+        else alert("Use valid email address, please.")
+     }
+
+     const handleSubmit = (e) => {
+        e.preventDefault();
+        if (user.password === user.passwordConfirm)
+        {console.log({user})
+        alert("Thanks for your registration.")}
+        else alert("Passwords do not match.")
+     }
+
+     return (
+        <div className='container'>
+        <h1>REGISTRATION</h1>
+        
+            
+
+        <form className='registration-form' onSubmit={(e) => handleSubmit(e)}>
+        <div className='outer-circle'>
+            <div className='inner-circle'></div>
+            </div>
+            <input type="email" placeholder="Email Address" autoFocus id="email" onChange={(e) => handleEmail(e.target.value)} />
```

Comment:
tyhle inputy nejsou "provázané" se stavem jak bylo v zadání. Slovem "provázaný" Luděk myslel že formulář má být kontrolovaný (https://reactjs.org/docs/forms.html#controlled-components) kdežto ty máš https://reactjs.org/docs/uncontrolled-components.html .

Rozdíl je v tom že v tvém případě si komponenta <input žije svým vlastním životem, má svůj vlastní stav a NAVÍC k tomu že má vlastní stav na onChange dává vědět tobě a ty si ten stav ukládáš znovu. Tím pádem je stav inputu de facto na 2 místech. Problém je že kdybys teď z nějakého místa zavolala `setUser({...user, email: "neco"})` tak se ti to ve formuláři neprojeví (projeví se to jen ve tvém stavu ale ten se do formuláře nijak nepropíše).

Prosím změň formulář tak aby byl kontrolovaný (tj každé políčko mělo atribut value= a správnou hodnotu ze stavu)

Reply:
ano, inputy jsem měla uncontrolled... původně totiž, když jsem je tam dávala jako controlled, vůbec mi nešlo do pole psát a nemohla jsem přijít na to proč, zazmatkovala jsem a předělala jsem to na uncontrolled, aby mi do pole psát šlo (měla jsem to rozhodně řešit jinak). Zkusila jsem to teď opravit a zjistila jsem, že jsem měla původně jen špatně napsaný handler. Už by to mělo být v pořádku.